packages/main/src/plugin/provider-impl.ts

```diff
@@ -75,6 +75,7 @@ export class ProviderImpl implements Provider, IDisposable {
 
   constructor(
     private _internalId: string,
+    readonly extension: string,
```

Comment:
missing private ?

Reply:
I would rename extension to extensionId if it contains the extension id

extension let me think we have access to all the attributes

Reply:
there is no `private` as Denis commented to only use readonly to avoid adding the get method https://github.com/containers/podman-desktop/pull/3680#discussion_r1306035693

Reply:
in that case we should probably add readonly in the interface ProviderInfo